src/components/numeral-date/numeral-date.spec.tsx

```diff
@@ -651,6 +651,29 @@ describe("NumeralDate", () => {
     });
   });
 
+  describe("optional", () => {
+    it("the isOptional prop is passed to the Textbox", () => {
+      const PropWrapper = shallow(
+        <NumeralDate value={{ dd: "12", mm: "", yyyy: "" }} isOptional />
+      );
+
+      const textbox = PropWrapper.find("Textbox").first();
+      expect(textbox.prop("isOptional")).toBe(true);
+    });
+
+    it("the isOptional prop is passed to the Textbox and set to false", () => {
+      const PropWrapper = shallow(
+        <NumeralDate
+          value={{ dd: "12", mm: "", yyyy: "" }}
+          isOptional={false}
+        />
+      );
+
+      const textbox = PropWrapper.find("Textbox").first();
+      expect(textbox.prop("isOptional")).toBeUndefined();
+    });
```

Comment:
**suggestion**: thanks for adding these tests @DobroTora.

As `NumeralDate` has three inputs instead of one, I think it would be good to test each of them to make sure they are all covered by tests. Would you be able to modify these two tests slightly?:

```ts
it("the isOptional prop is passed to the Textbox", () => {
  const PropWrapper = mount(
    <NumeralDate value={{ dd: "12", mm: "", yyyy: "" }} isOptional />
  );

  const textbox = PropWrapper.find("Textbox");
  textbox.forEach((node) => {
    expect(node.prop("isOptional")).toBeTruthy();
  });
});

it("the isOptional prop is passed to the Textbox and set to false", () => {
  const PropWrapper = mount(
    <NumeralDate
      value={{ dd: "12", mm: "", yyyy: "" }}
      isOptional={false}
    />
  );

  const textbox = PropWrapper.find("Textbox");
  textbox.forEach((node) => {
    expect(node.prop("isOptional")).toBeUndefined();
  });
});
```

Reply:
I actually think we shouldn't pass [isOptional](https://github.com/Sage/carbon/blob/645b53b9953a45d26013d0b4a8b769761c9341b1/src/components/numeral-date/numeral-date.component.tsx#L509) to the individual inputs here as all we need is it append to the label which it does etc. I think if we remove this line the above test won't work, we should do the same test as we did for the Selects where we assert `(optional)` is added `::after` the label imo

```
   assertStyleMatch(
        {
          content: '"(optional)"',
          fontWeight: "350",
          marginLeft: "4px",
        },
        wrapper.find(StyledLabelContainer),
        { modifier: "::after" }
      );